plutus-use-cases/src/Language/PlutusTx/Coordination/Contracts/TokenAccount.hs

```diff
@@ -0,0 +1,181 @@
+
+import qualified Language.Plutus.Contract.Typed.Tx                 as TypedTx
+import           Ledger                                            (Address, PubKey, TxOut)
+import qualified Ledger                                            as Ledger
+import           Ledger.TxId                                       (TxId)
+import           Ledger.Typed.Scripts                              (ScriptType (..))
+import qualified Ledger.Typed.Scripts                              as Scripts
+import qualified Ledger.Validation                                 as V
+import           Ledger.Value                                      (CurrencySymbol, TokenName, Value)
+import qualified Ledger.Value                                      as Value
+
+import qualified Language.PlutusTx.Coordination.Contracts.Currency as Currency
+
+newtype AccountOwner = AccountOwner { unAccountOwner :: (CurrencySymbol, TokenName) }
+    deriving newtype (Eq, Show)
+
+data TokenAccount
+
+instance ScriptType TokenAccount where
+    type RedeemerType TokenAccount = ()
+    type DataType TokenAccount = AccountOwner
+
+type TokenAccountSchema =
+    BlockchainActions
+        .\/ Endpoint "redeem" (AccountOwner, PubKey)
+        .\/ Endpoint "pay" (AccountOwner, Value)
+        .\/ Endpoint "new-account" (TokenName, PubKey)
+
+type HasTokenAccountSchema s =
+    ( HasBlockchainActions s
+    , HasEndpoint "redeem" (AccountOwner, PubKey) s
+    , HasEndpoint "pay" (AccountOwner, Value) s
+    , HasEndpoint "new-account" (TokenName, PubKey) s
+    )
+
+-- | 'transfer', 'redeem', 'pay' and 'newAccount' with endpoints.
+tokenAccountContract
+    :: forall s e.
+       ( HasTokenAccountSchema s
+       , AsContractError e
+       )
+    => Contract s e ()
+tokenAccountContract = redeem_ <|> pay_ <|> newAccount_ where
+    redeem_ = do
+        (accountOwner, destination) <- endpoint @"redeem" @(AccountOwner, PubKey) @s
+        void $ redeem destination accountOwner
+        tokenAccountContract
+    pay_ = do
+        (accountOwner, value) <- endpoint @"pay" @_ @s
+        void $ pay accountOwner value
+        tokenAccountContract
+    newAccount_ = do
+        (tokenName, initialOwner) <- endpoint @"new-account" @_ @s
+        void $ newAccount tokenName initialOwner
+        tokenAccountContract
+
+{-# INLINEABLE accountToken #-}
+accountToken :: AccountOwner -> Value
+accountToken (AccountOwner (symbol, name)) = Value.singleton symbol name 1
+
+{-# INLINEABLE validate #-}
+validate :: AccountOwner -> () -> V.PendingTx -> Bool
+validate owner _ ptx = V.valueSpent ptx `Value.geq` accountToken owner
+
+scriptInstance :: Scripts.ScriptInstance TokenAccount
+scriptInstance =
+    Scripts.Validator @TokenAccount $$(PlutusTx.compile [|| validate ||]) $$(PlutusTx.compile [|| wrap ||])
+    where
+    wrap = Scripts.wrapValidator @AccountOwner @()
+
+address :: Address
+address = Scripts.scriptAddress scriptInstance
+
+-- | Pay some money to the given token
```

Comment:
```suggestion
-- | Pay some money to the given token account
```

Reply:
This is slightly confusing given that the type is `AccountOwner` not `Account`, but it's clear enough.